services/kvstoremanager/redis.go

```diff
@@ -150,3 +168,60 @@ func (m *redisManagerT) HSet(hash, key string, value interface{}) (err error) {
 	}
 	return err
 }
+
+func (m *RedisManager) ExtractJSONSetArgs(jsonData json.RawMessage) ([]string, error) {
+	key := gjson.GetBytes(jsonData, "message.key").String()
+	path := gjson.GetBytes(jsonData, "message.path").String()
+	jsonVal := gjson.GetBytes(jsonData, "message.value")
+
+	actualPath := "$" // root insert
+	if path != "" {
+		actualPath = fmt.Sprintf("$.%s", path)
+	}
+	args := []string{key, actualPath, jsonVal.String()}
+
+	if actualPath != "$" {
+		v, err := m.GetClient().JSONGet(context.Background(), key).Result()
+		if err != nil {
+			return nil, err
+		}
+		if v == "" {
+			// key is new one but we need to insert a value other than root
+			// formulate {[path]: value}
+			mapStr, err := json.Marshal(map[string]interface{}{
+				path: jsonVal.Value(),
+			})
+			if err != nil {
+				return nil, err
+			}
+			// data is not present in key
+			args = []string{key, "$", string(mapStr)} // arguments to insert data
+		}
+	}
+	return args, nil
+}
+
+func (m *RedisManager) SendDataAsJSON(jsonData json.RawMessage) (interface{}, error) {
+	nmSetArgs, err := m.ExtractJSONSetArgs(jsonData)
+	if err != nil {
+		return nil, err
+	}
+	redisClient := m.GetClient()
+	ctx := context.Background()
+	val, err := redisClient.JSONSet(ctx, nmSetArgs[0], nmSetArgs[1], nmSetArgs[2]).Result()
+	if err != nil {
+		return nil, fmt.Errorf("setting key:(%s %s %s): %w", nmSetArgs[0], nmSetArgs[1], nmSetArgs[2], err)
+	}
+
+	return val, err
+}
+
+func (*RedisManager) ShouldSendDataAsJSON(config map[string]interface{}) bool {
+	var dataAsJSON bool
+	if dataAsJSONI, ok := config["useJSONModule"]; ok {
+		if dataAsJSON, ok = dataAsJSONI.(bool); ok {
+			return dataAsJSON
+		}
+	}
+	return dataAsJSON
+}
```

Comment:
Why is this in the `RedisManager`? It doesn't seem to be related to Redis operations specifically. In fact it does not need a client and there is no struct receiver. The only dependency is the configuration which is passed as a parameter.

I suggest moving this elsewhere. The custom destination manager seems to be a good candidate. Thoughts?